sql/expression/function/date.go

```diff
@@ -489,3 +496,65 @@ func (c CurrDate) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 func (c CurrDate) WithChildren(children ...sql.Expression) (sql.Expression, error) {
 	return NoArgFuncWithChildren(c, children)
 }
+
+// Determines the return type of a DateAdd/DateSub expression
+// Logic is based on https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-add
+func getType(input sql.Expression, interval *expression.Interval) sql.Type {
+	inputType := input.Type()
```

Comment:
If input is nil this is a null pointer exception

Reply:
What's the validation approach we should be following? `Eval` is using the same data (`d.Date` and `d.Interval`) without checking them first.
And in general I'm not seeing a lot of nil checks under `sql/expression/function/`, looks like we validate the input data when constructing the objects, then just trust the properties to contain valid data.

Reply:
Generally speaking every `sql.Expression` can be `nil`, and function implementations need to check for that and do any appropriate null handling